src/components/Graph.react.js

```diff
@@ -73,208 +54,92 @@ const filterEventData = (gd, eventData, event) => {
-
-        const gd = this.gd.current;
-
-        gd.on('plotly_click', eventData => {
-            const clickData = filterEventData(gd, eventData, 'click');
-            if (!isNil(clickData)) {
-                setProps({clickData});
-            }
-        });
-        gd.on('plotly_clickannotation', eventData => {
-            const clickAnnotationData = omit(
-                ['event', 'fullAnnotation'],
-                eventData
-            );
-            setProps({clickAnnotationData});
-        });
-        gd.on('plotly_hover', eventData => {
-            const hover = filterEventData(gd, eventData, 'hover');
-            if (!isNil(hover) && !equals(hover, hoverData)) {
-                setProps({hoverData: hover});
-            }
-        });
-        gd.on('plotly_selected', eventData => {
-            const selected = filterEventData(gd, eventData, 'selected');
-            if (!isNil(selected) && !equals(selected, selectedData)) {
-                setProps({selectedData: selected});
-            }
-        });
-        gd.on('plotly_deselect', () => {
-            setProps({selectedData: null});
-        });
-        gd.on('plotly_relayout', eventData => {
-            const relayout = filterEventData(gd, eventData, 'relayout');
-            if (!isNil(relayout) && !equals(relayout, relayoutData)) {
-                setProps({relayoutData: relayout});
-            }
-        });
-        gd.on('plotly_restyle', eventData => {
-            const restyle = filterEventData(gd, eventData, 'restyle');
-            if (!isNil(restyle) && !equals(restyle, restyleData)) {
-                setProps({restyleData: restyle});
-            }
-        });
-        gd.on('plotly_unhover', () => {
-            if (clear_on_unhover) {
-                setProps({hoverData: null});
-            }
-        });
-    }
+    clearExtendData() {
+        this.setState(({extendData}) => {
+            const res =
+                extendData && extendData.length
+                    ? {
+                          extendData: EMPTY_EXTEND_DATA,
+                      }
+                    : undefined;
 
-    componentDidMount() {
-        this.plot(this.props).then(() => {
-            window.addEventListener('resize', this.graphResize);
+            return res;
         });
     }
 
-    componentWillUnmount() {
-        const gd = this.gd.current;
-        if (gd && gd.removeAllListeners) {
-            gd.removeAllListeners();
-            Plotly.purge(gd);
-        }
-        window.removeEventListener('resize', this.graphResize);
-    }
-
-    shouldComponentUpdate(nextProps) {
+    render() {
         return (
-            this.props.id !== nextProps.id ||
-            JSON.stringify(this.props.style) !== JSON.stringify(nextProps.style)
+            <ControlledPlotlyGraph
+                {...{
+                    ...this.props,
+                    extendData: this.state.extendData,
+                    clearExtendData: this.clearExtendData,
```

Comment:
When the real component is done processing `extendData` entries, it calls `clearExtendData`